src/NServiceBus.Persistence.AzureTable/SynchronizedStorage/TableClientHolder.cs

```diff
@@ -1,11 +1,30 @@
 ﻿namespace NServiceBus.Persistence.AzureTable
 {
+    using System;
+    using System.Collections.Concurrent;
+    using System.Threading;
+    using System.Threading.Tasks;
     using Azure.Data.Tables;
 
     sealed class TableClientHolder
     {
-        public TableClientHolder(TableClient tableClient) => TableClient = tableClient;
+        static readonly ConcurrentDictionary<string, bool> createdTables = new();
```

Comment:
A few thoughts. Theoretically the dictionary could have been instance based on the persistence storage type since that would prevent tables caching between multiple endpoints in the same process.

TableHolder was thought to be a per pipeline state holder and now it is also a table creation cache. How about we keep the holder a simple holder and extract all the table creation and caching into a dedicated type that also gets the settings. Then we have all table creation in one place.

That type can be a singleton shared between the saga and outbox storages. Then we can also make the cache non static

Reply:
The drawback of that approach might the management of the table creation settings per storage type if we need thag